src/spu/interface.h

```diff
@@ -91,8 +91,37 @@ class impl final : public SPUInterface {
     static const size_t NSSIZE = 45;
 
     // spu
+    class main_t_variables {
```

Comment:
What's the reasoning behind shoveling all of those stack variables into a member that'll be allocated on the heap? The compiler will no longer be able to hoist those into local registers, and this makes the code less readable overall.

Also the reset function is called exactly once during the whole execution, making it fairly pointless, while mixing self initialization of some variables in, but not others.

Reply:
**_What's the reasoning behind shoveling all of those stack variables into a member that'll be allocated on the heap?_**
The main reason was that, for breaking the main thread into manageable methods, there are 2 options:

1. Passing local variables to calling functions:
 + it's faster
 - it's less readable
 - it's less scalable 

2. Hosting variables in a shared memory address:
  + common playground (shared memory address)
  + more scaleable and readable
  + more OOP (taking benefit of C++ features)
  - slower than the stack memory
so its a trade-off between scalability and speed

**_Also the reset function is called exactly once during the whole execution, making it fairly pointless_**

in some cases where we need to restart the main thread such as encountering an exception or malfunction

Reply:
I found a better way that satisfies both 2 options. that is, using a `struct` instead of a `class`, instantiating struct as a local variable, and passing the `struct`'s ref into the methods.